### PR TITLE
Added header to bookmarks panel

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/32952

Bookmarks panel's header includes icon, title and button for loading bookmarks manager.

`BraveBookmarksSidePanelCoordinator` gives bookmaks panel's contents view and that contents view
has our header view and upstream web view.

image

image

Resolves 

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Toggle bookmarks panel and check header is shown.
2. Click button in the header and check bookmarks manager is loaded in a tab

### CODE REVIEW

File: browser/ui/views/side_panel/bookmarks/brave_bookmarks_side_panel_coordinator.cc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+void BraveBookmarksSidePanelCoordinator::CreateAndRegisterEntry(
+    SidePanelRegistry* global_registry) {
+  global_registry->Register(std::make_unique<SidePanelEntry>(
+      SidePanelEntry::Id::kBookmarks,
+      l10n_util::GetStringUTF16(IDS_BOOKMARK_MANAGER_TITLE),
+      ui::ImageModel::FromVectorIcon(omnibox::kStarIcon, ui::kColorIcon),
+      base::BindRepeating(
+          &BraveBookmarksSidePanelCoordinator::CreateBookmarksPanelView,
+          base::Unretained(this))));
```
[the remark this answers is not in the document]
Comment (simonhong) [reply browser/ui/views/side_panel/bookmarks/brave_bookmarks_side_panel_coordinator.cc:33]: This `BraveBookmarksSidePanelCoordinator`  and `global_registry_` is per browser object. Both life cycle is tied with Browser object. So, both are destroyed together when `Browser` object is deleted. So, I think  using `Unretained()` here would not cause any issue. Also all upstream coordinator uses this pattern.

File: browser/ui/views/side_panel/brave_bookmarks_side_panel_view.cc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    spacer->SetProperty(
+        views::kFlexBehaviorKey,
+        views::FlexSpecification(views::MinimumFlexSizeRule::kScaleToZero,
+                                 views::MaximumFlexSizeRule::kUnbounded)
+            .WithOrder(2));
+    // Safe to use Unretained(this) here as |button| will be destroyed before
+    // this class.
+    auto* button = AddChildView(std::make_unique<views::ImageButton>(
+        base::BindRepeating(&BookmarksSidePanelHeaderView::OnButtonPressed,
+                            base::Unretained(this), browser)));
```
[the remark this answers is not in the document]
Comment (simonhong) [reply browser/ui/views/side_panel/brave_bookmarks_side_panel_view.cc:79]:
```
// Safe to use Unretained(this) here as |button| will be destroyed before
// this class.
```

File: browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "base/functional/callback_forward.h"
+#include "base/scoped_observation.h"
+#include "ui/views/view.h"
+#include "ui/views/view_observer.h"
+
+class Browser;
+
+// Gives bookmarks panel specific header view with web view.
+class BraveBookmarksSidePanelView : public views::View,
```
Comment (sangwoo108) [inline browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h:17]: optional: I'm wondering if it's worth making a base class for this and ReadLaterSidePanelView. They're likely to be updated together in terms of styling, so generalizing them could be useful I guess. But I'm okay if you're leaving this as is.

File: browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                                    public views::ViewObserver {
+ public:
+  explicit BraveBookmarksSidePanelView(Browser* browser);
+  ~BraveBookmarksSidePanelView() override;
+  BraveBookmarksSidePanelView(const BraveBookmarksSidePanelView&) = delete;
+  BraveBookmarksSidePanelView& operator=(const BraveBookmarksSidePanelView&) =
+      delete;
+
+ private:
+  void OnViewVisibilityChanged(views::View* observed_view,
```
Comment (sangwoo108) [inline browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h:27]: nit:
```suggestion
  // views::ViewObserver:
  void OnViewVisibilityChanged(views::View* observed_view,
```

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h (hunk 4 above)
Comment (simonhong) [reply browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h]: Done.

File: browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h (hunk 3 above)
Comment (simonhong) [reply browser/ui/views/side_panel/brave_bookmarks_side_panel_view.h]: good call. `BraveSidePanelViewBase` is introduced.

**Review by sangwoo108 — APPROVED** [review]
Still LGTM

**Review by goodov — APPROVED** [review]
`chromium_src` lgtm

**Review by mkarolin — APPROVED** [review]
`strings`++

File: app/theme/default_100_percent/brave/sidebar_bookmarks_panel_header.png
Comment (bridiver) [inline app/theme/default_100_percent/brave/sidebar_bookmarks_panel_header.png:1]: these files are getting copied into `chrome/app/theme/default_100_percent/chromium` which doesn't seem correct.

**Pull request merged**